types/react-cropper/index.d.ts

```diff
@@ -11,7 +11,7 @@ import * as React from 'react';
 type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
 
 export interface ReactCropperProps extends Cropper.Options, Omit<React.HTMLProps<HTMLImageElement>, 'data' | 'ref'> {
-    ref?: string | ((cropper: null | ReactCropper) => any);
+    ref?: string | React.RefObject<Cropper | undefined> | ((cropper: null | ReactCropper) => any);
```

Comment:
@frisi I don't known if `Cropper | undefined` is accurate:
- on the right we are using `ReactCropper` instead of `Cropper`
- when using this ref I expect to have a null value or a `ReactCropper` type; undefined is not accurate here

At this point I suggest to modify the interface like this:
```
interface ReactCropperProps_ extends
  Cropper.Options, Omit<React.HTMLProps<HTMLImageElement>, 'data' | 'ref'>,
  React.RefAttributes<ReactCropper> {}
```
In this way we are reusing the `React.RefAttributes` instead of enumerating one by one like we are doing in this case.
Do you agree?

Reply:
sure - agreed. thanks for your reply and proposal @bwlt !

did not know about these interfaces - much cleaner this way.
however, the `LegacyRef` (string) will not be supported this way.
dunno if somebody else still needs it, but we could use that to be sure this usecase also works.

on the other side people using older react-versions that require refs as string (i guess this has something to do with the react version) can still use an older version of @types/react-cropper

``` 
    type Ref<T> = RefCallback<T> | RefObject<T> | null;
    type LegacyRef<T> = string | Ref<T>;

   interface RefAttributes<T> extends Attributes {
        ref?: Ref<T>;
    }
    interface ClassAttributes<T> extends Attributes {
        ref?: LegacyRef<T>;
    }
```

i'll update the pr and hope you can merge and release soon ;-)

Reply:
Maybe it is better to maintain the retrocompatibility instead. The LegacyRef is more appropriate in this case

Reply:
i also chose to do still support it as there was also a tests for that.
added a test for `FunctionComponents` using `useRef` hook too.

is there anything else to do for me or can you take care of a release @bwlt ?

Reply:
I'm only a reviewer :)
The ref related changes looks good to me, I don't know the impact for the other changes like this: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42801/files#diff-e7572847de7a9e112a5be7b0a58c7c31R7 .
Better to await another reviewer

Reply:
the update of the minimal typescript version was necessary to make tests pass and was suggested by the test output:

```
/sandbox/DefinitelyTyped/types/react-cropper/react-cropper-tests.tsx:37:37
ERROR: 37:37  expect     Compile error in typescript@3.6 but not in typescript@3.7.
Fix with a comment '// Minimum TypeScript Version: 3.7' just under the header.
Expression expected.
ERROR: 37:51  expect     Compile error in typescript@3.6 but not in typescript@3.7.
Fix with a comment '// Minimum TypeScript Version: 3.7' just under the header.
':' expected.
```

so i'll wait for another review then.... have a nice weekend!